### PR TITLE
Removing flag for EIP1559 V2 support

### LINKED ISSUE #14925: Remove Enhanced Gas UI from Experimental Flag
### Describe the bug

Remove Enhanced Gas UI from experimental flag by

- Enable Enhanced Gas (EIP-1559 v2) UI for transaction screens by default
- Removing it from the Settings > Experimental
- Remove Enhanced Gas UI alert that appears if the user is on EiP-1559 v1 UI

### Steps to reproduce

N/A

### Error messages or log output

_No response_

### Version

10.14.6

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jpuri**
Fixes: #14925
Removing `eip1559V2Enabled` flag.

### CODE REVIEW

**Comment by jpuri** [comment]
Hey @seaona : this PR requires QA for both legacy and EIP-1559 transactions.

**Comment by segun** [comment]
This is a huge change, I got lost in ui/components/app/edit-gas-display/edit-gas-display.component.js :(

Any chance we can jump on a call together so you can give me a bit more context.

Thanks.

**Review by darkwing — COMMENTED** [review]
Code looks good but we need a rebase.

**Review by brad-decker — DISMISSED** [review]

**Comment by jpuri** [comment]
Updating the branch is has broken the build. I will fix that and do more testing... bit of pain to update huge PR like this :)

**Comment by seaona** [comment]
Started QAing the PR. Will add some comments on the way.

On Legacy Transactions, the Suggested Gas fee is not displayed anymore.

**Repro Steps:**
1. Go to Test Dapp https://metamask.github.io/test-dapp/
2. Connect MM
3. Click Send Legacy Transacion
4. Click Edit -- see how no Suggested Gas is displayed
5. Click Edit suggested gas fee -- see how no Suggested Gas is displayed

**Result**:

image

image

**Expected**:
image

image

**Comment by jpuri** [comment]
@seaona : I fixed the issue.

**Review by brad-decker — DISMISSED** [review]

File: test/e2e/tests/send-eth.spec.js [hunk 1]
```diff
@@ -251,25 +251,13 @@ describe('Send ETH from dapp using advanced gas controls', function () {
         );
         await driver.assertElementNotPresent({ text: 'Data', tag: 'li' });
         await driver.clickElement({ text: 'Edit', tag: 'button' });
-        await driver.waitForSelector({
```
Comment (digiwand) [inline test/e2e/tests/send-eth.spec.js:254]: should we keep one of these `.transaction-total-banner` checks to verify the value still exists?

**Review by brad-decker — DISMISSED** [review]

File: test/e2e/tests/send-eth.spec.js (hunk 1 above)
Comment (jpuri) [reply test/e2e/tests/send-eth.spec.js:254]: done

File: ui/components/app/edit-gas-display/edit-gas-display.component.js [hunk 2]
```diff
@@ -1,104 +1,47 @@
-import React, { useContext, useLayoutEffect, useRef, useState } from 'react';
-import { useSelector } from 'react-redux';
+import React, { useContext, useRef } from 'react';
```
Comment (darkwing) [inline ui/components/app/edit-gas-display/edit-gas-display.component.js:1]: Is everything removed in this file really meant to go?  Do we not need `supportsEIP1559` checks anymore?

Comment (jpuri) [reply ui/components/app/edit-gas-display/edit-gas-display.component.js:1]: We now get rid of EIP-1559 V1 implementation so this file will be used for only legacy transactions. For EIP-1559 now V2 implementation will be the default.

**Review by pedronfigueiredo — APPROVED** [review]

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by brad-decker — DISMISSED** [review]

**Comment by seaona** [comment]
I've re-tested and looks good on my end @jpuri

**Review by brad-decker — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**